### PR TITLE
Added support for indicating directionality on Graphs

### PR DESCRIPTION
**Pull request by philippjfr**
Adds support for arrows on Graph edges to allow indicating directionality of graphs. uses same trick as VectorField to render the arrows.

- [x] Closes https://github.com/ioam/holoviews/issues/2521
- [x] Add gallery example
- [x] Add section to networks user guide

Simple example:

bokeh_plot

Real data example:

bokeh_plot

### CODE REVIEW

File: holoviews/plotting/bokeh/graphs.py [hunk 1]
```diff
@@ -12,16 +12,24 @@
...
-from ..util import process_cmap
+from ..util import process_cmap, get_directed_graph_paths
 from .util import rgba_tuple
 
 
 class GraphPlot(CompositeElementPlot, ColorbarPlot, LegendPlot):
 
+    arrow_length = param.Number(default=0.025, doc="""
+      If directed option is enabled this determines the length of the
+      arrows as fraction of the overall extent of the graph.""")
```
Comment (jlstevens) [inline holoviews/plotting/bokeh/graphs.py]: The size of the arrow head I assume? I would expect the length of the arrow to be the length of the edge...

Comment (jlstevens) [reply holoviews/plotting/bokeh/graphs.py]: If this is what you mean, then I'm not so sure about the name...

Comment (philippjfr) [reply holoviews/plotting/bokeh/graphs.py]: True, it's the arrow head, ``head_length`` doesn't seem that descriptive and ``arrow_head_length`` seems too long. Any other suggestions?

Comment (jlstevens) [reply holoviews/plotting/bokeh/graphs.py]: Maybe `size` as it is shorter than `length`?

Comment (jlstevens) [reply holoviews/plotting/bokeh/graphs.py]: I also think `arrow_size` is more obviously associated with the head of the arrow. I think `arrow_size` might be shorter and less confusing...

Comment (jbednar) [reply holoviews/plotting/bokeh/graphs.py]: I don't associate `arrow_size` with the head; that seems like the size of some arrow glyph.

File: examples/gallery/demos/bokeh/directed_airline_routes.ipynb [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "as_graph.options(\n",
+    "    directed=True, node_size=8, padding=0.1, bgcolor='gray', xaxis=None, yaxis=None,\n",
+    "    edge_line_color='white', edge_line_width=1, width=800, height=800, arrow_length=0.01,\n",
```
Comment (philippjfr) [inline examples/gallery/demos/bokeh/directed_airline_routes.ipynb]:
```suggestion
    "    edge_line_color='white', edge_line_width=1, width=800, height=800, arrowhead_length=0.01,\n",
```

File: examples/gallery/demos/matplotlib/directed_airline_routes.ipynb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "as_graph.options(\n",
+    "    directed=True, node_size=8, padding=0.1, bgcolor='gray', xaxis=None, yaxis=None,\n",
+    "    edge_color='white', edge_linewidth=1, arrow_length=0.01,\n",
```
Comment (philippjfr) [inline examples/gallery/demos/matplotlib/directed_airline_routes.ipynb]:
```suggestion
    "    edge_color='white', edge_linewidth=1, arrowhead_length=0.01,\n",
```

File: examples/user_guide/Network_Graphs.ipynb [hunk 4]
```diff
@@ -70,6 +70,31 @@
...
+    "When specifying the graph edges the source and target node are listed in order, if the graph is actually a directed graph this may used to indicate the directionality of the graph. By setting ``directed=True`` as a plot option it is possible to indicate the directionality of each edge using an arrow:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "simple_graph.options(directed=True, node_size=5, arrow_length=0.05)"
```
Comment (philippjfr) [inline examples/user_guide/Network_Graphs.ipynb]:
```suggestion
    "simple_graph.options(directed=True, node_size=5, arrowhead_length=0.05)"
```

File: examples/user_guide/Network_Graphs.ipynb [hunk 5]
```diff
@@ -70,6 +70,31 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The length of the arrows can be set as an fraction of the overall graph extent using the ``arrow_length`` option."
```
Comment (philippjfr) [inline examples/user_guide/Network_Graphs.ipynb]:
```suggestion
    "The length of the arrows can be set as an fraction of the overall graph extent using the ``arrowhead_length`` option."
```

**Comment by philippjfr** [comment]
Done the requested renaming so I'll go ahead and merge.

**Pull request merged**